### PR TITLE
Manage Forma.FP project and enhance Result/Option functionalities

### PR DESCRIPTION
**Pull request by pietroserrano**
- [x] Add meaningful XML documentation to all empty `<summary>`, `<typeparam>`, `<param>`, and `<returns>` tags in `ResultAsyncExtensions.cs`
- [x] Fix `docs/guides/web-api.md` to use `Result<T>` + `Error.*` throughout `ProductService` and `OrderService`
- [x] Fix `docs/packages/fp.md`: add `async` keyword to `FetchUserAsync` method signature
- [x] Fix `docs/packages/core.md`: update `Result<TSuccess, TFailure>` references to use `Result<T>` + `Error`
- [x] Fix `docs/packages/fp.md`: replace raw string arguments in `Failure(...)` calls with proper `Error.*` factory calls
- [x] Fix `docs/packages/fp.md`: make `ApiError` record extend `Error`
- [x] Fix `examples/console/Forma.Examples.Console.FP/README.md`: update `Result<TSuccess, TFailure>` reference
- [x] Fix `src/Forma.Core/FP/ResultExtensions.cs`: `ValidateAll` now fails fast on non-`ValidationError` failures instead of silently dropping them
- [x] Fix `examples/console/Forma.Examples.Console.FP/Forma.Examples.Console.FP.csproj`: use `$(UseProjectReferences)` pattern; update `Microsoft.Extensions.*` from 9.0.7 to 10.0.5
- [x] Fix `src/Forma.Benchmarks/FPBenchmarks.cs`: replace local `count` variables in `Do`/`OnSuccess`/`OnError` benchmarks with observable instance field `_sideEffectCounter`
- [x] Fix `docs/packages/fp.md`: both `ValidateAsync` snippets now use `Error.Generic` instead of a string factory
- [x] Fix `docs/guides/web-api.md`: replace all `Result<T, string>` with `Result<T>` + `Error.*`; fix `onSome`/`onNone` → `some`/`none`; fix string `Failure` calls in `CreateUserCommandHandler`
- [x] Fix `docs/guides/console-app.md`: fix `.Then(content => content.ToUpper())` → `.Then(content => Result<string>.Success(content.ToUpper()))`; fix `onSome`/`onNone` → `some`/`none`; fix string `Failure` calls; fix `Result<int, string>` quickstart example; use `error.Message` in `onFailure` interpolations; fix `// Output:` comment for `DataFormat` error message shape
- [x] Fix `docs/packages/fp.md`: rewrite form validation example to use `Error.Validation(...)` instead of a custom local record and `List<ValidationError>`; fix Match `onFailure` to use `ValidationError` pattern match
- [x] Add `IsFailure` convenience property (`=> !IsSuccess`) to `Result<T>` for symmetry with `IsSuccess`
- [x] Add null guard in `Result<T>.Success()` — throws `ArgumentNullException` if value is null, enforcing the success invariant at runtime
- [x] Add null guard in `Option<T>.Some()` — throws `ArgumentNullException` if value is null, preserving the invariant that `Some` always wraps a usable value
- [x] Make `ValidationError` and `AggregateError` truly immutable: constructor defensively copies the incoming dictionary (and each `string[]` array) and error list into read-only collections
- [x] Add `ArgumentNullException.ThrowIfNull` guards to `ResultExtensions.Try` / `TryAsync` for the `func` parameter
- [x] Add `ArgumentNullException.ThrowIfNull` guards to `ResultExtensions.Recover` for the `recovery` parameter and `OrElseTry` for the `alternative` parameter
- [x] Add `ArgumentNullException.ThrowIfNull` guards to all `OptionAsyncExtensions` methods (`ThenAsync`, `DoAsync`, `ValidateAsync`) for the task and delegate parameters
- [x] Add `ArgumentNullException.ThrowIfNull` guards to `ResultAsyncExtensions.ThenAsync` (`next`), `DoAsync` (`resultTask`, `action`), `ValidateAsync` (`resultTask`, `predicate`, `errorFactory`), and `MatchAsync` (`resultTask`, `onSuccess`, `onFailure`)
- [x] Optimize `Error.Metadata` to cache the `ReadOnlyDictionary` wrapper once in `init` (stored in `_metadataReadOnly`) instead of allocating a new wrapper on every getter call
- [x] Fix `ErrorExtensions.ValidationError.Combine` to clone `string[]` arrays from both source errors when building the combined dictionary, preventing sharing of mutable array references
- [x] Rename `Validation(params ...)` factory tuple field `Error` → `Message` for clarity
- [x] Add unit tests for `ResultExtensions`: `Try`, `TryAsync`, `Recover`, `OrElse`, `OrElseTry`, `Tap`, `ValidateAll` (accumulation and fail-fast paths), `Combine` (2- and 3-result overloads), `IsFailure`, null guard in `Success()`, null guard in `Try`/`TryAsync`/`Recover`/`OrElseTry`, and defensive copy behaviour of `ValidationError`/`AggregateError`
- [x] Add unit tests for `ErrorExtensions.ValidationError.Combine`: array-cloning tests for mutation safety and shared-key merging
- [x] Add unit tests for `OptionAsyncExtensions`: `ThenAsync`, `DoAsync`, and `ValidateAsync` (Some/None propagation, binder/action not invoked on None, null-guard checks)
- [x] Add unit tests for `ResultAsyncExtensions`: null-guard tests for `ThenAsync`, `DoAsync`, `ValidateAsync`, and `MatchAsync`
- [x] Add null-guard test for `Option<T>.Some(null!)`

### CODE REVIEW

File: src/Forma.Core/FP/ResultAsyncExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <typeparam name="TSuccess"></typeparam>
+    /// <typeparam name="TError"></typeparam>
+    /// <typeparam name="TNextSuccess"></typeparam>
+    /// <param name="resultTask"></param>
+    /// <param name="next"></param>
+    /// <returns></returns>
```
[the remark this answers is not in the document]
Comment (pietroserrano) [reply src/Forma.Core/FP/ResultAsyncExtensions.cs]: @copilot apply changes based on [this feedback](https://github.com/pietroserrano/forma/pull/44#discussion_r3017809168)

**Comment by pietroserrano** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/pietroserrano/forma/pull/44#pullrequestreview-4040315643)

**Comment by pietroserrano** [comment]
@copilot+claude-sonnet-4.6 apply changes based on the comments in [this thread](https://github.com/pietroserrano/forma/pull/44#pullrequestreview-4040407194)

**Comment by pietroserrano** [comment]
@copilot+claude-sonnet-4.6 apply changes based on the comments in [this thread](https://github.com/pietroserrano/forma/pull/44#pullrequestreview-4043395217)
